test/integration/monitoring-alert/monitoring_alert_test.go

```diff
@@ -0,0 +1,131 @@
+				if strings.Contains(logMessage, expectedLogMessage) && strings.Contains(logName, expectedLogName) {
+					// Test succeded.
+					return false, nil
+				}
+			}
+			return true, errors.New("Alert wasn't fired correctly.")
+		}, 200, 10*time.Second)
+	})
+	secretT.Test()
+}
+
+func TestMonitoringAlertAllSecrets(t *testing.T) {
+	email_addresses := []string{"email@example.com", "email2@example.com"}
+	vars := map[string]interface{}{
+		"email_addresses":     email_addresses,
+		"monitor_all_secrets": true,
+	}
+
+	secretT := tft.NewTFBlueprintTest(t, tft.WithVars(vars))
+
+	secretT.DefineVerify(func(assert *assert.Assertions) {
+		secretT.DefaultVerify(assert)
+		projectId := secretT.GetStringOutput("project_id")
+		projectDescribe := gcloud.Runf(t, "projects describe %s", projectId)
+		projectNumber := projectDescribe.Get("projectNumber").String()
+		secretDescribe := gcloud.Runf(t, "secrets describe %s --project %s", "secret-1", projectId)
+		secretName := secretDescribe.Get("name").String()
+		assert.Equal(fmt.Sprintf("projects/%s/secrets/secret-1", projectNumber), secretName, "has expected name")
+
+		notificationChannelNames := secretT.GetJsonOutput("notification_channel_names").Array()
+		assert.Len(notificationChannelNames, 2)
+		notificationChannelName := notificationChannelNames[0].String()
+		monitoringChannel := gcloud.Runf(t, "beta monitoring channels list --project %s --filter name='%s'", projectId, notificationChannelName).Array()
+		assert.Len(monitoringChannel, 1)
+		notificationChannelName2 := notificationChannelNames[1].String()
+		monitoringChannel2 := gcloud.Runf(t, "beta monitoring channels list --project %s --filter name='%s'", projectId, notificationChannelName2).Array()
+		assert.Len(monitoringChannel2, 1)
+		assert.ElementsMatch(email_addresses, []string{monitoringChannel[0].Get("labels.email_address").String(), monitoringChannel2[0].Get("labels.email_address").String()})
+
+		monitoringAlerts := gcloud.Runf(t, "alpha monitoring policies list --project %s", projectId).Array()
+		assert.Len(monitoringAlerts, 1)
+		monitoringAlert := monitoringAlerts[0]
+		alertCondition := monitoringAlerts[0].Get("conditions").Array()
+		assert.Len(alertCondition, 1)
+		expectedFilter := "protoPayload.methodName=\"google.cloud.secretmanager.v1.SecretManagerService.DestroySecretVersion\""
+		assert.Equal(expectedFilter, alertCondition[0].Get("conditionMatchedLog.filter").String())
+		notificationChannels := monitoringAlert.Get("notificationChannels").Array()
+		assert.Len(notificationChannels, 2)
+		assert.ElementsMatch([]string{notificationChannelName, notificationChannelName2}, []string{notificationChannels[0].String(), notificationChannels[1].String()})
+		assert.Equal("WARNING", monitoringAlert.Get("severity").String())
+		assert.Equal("300s", monitoringAlert.Get("alertStrategy.notificationRateLimit.period").String())
+		assert.True(monitoringAlert.Get("enabled").Bool())
+	})
+	secretT.Test()
+}
```

Comment:
is there a test needed for the boolean input if set to value other than the default?

Reply:
Yeah, the first test leaves the boolean input as default (false). The second test sets the boolean input as true with the goal to make sure the resources are correctly created when the boolean is alternated.

Reply:
Agree.